### ISSUE TITLE
Generated Swagger spec with required fields causes error on GCloud service-management deploy

### ISSUE THREAD

**Issue by Synthoova** [cloudendpoints/endpoints-python#21]
I've noticed something with the google-endpoints package, which I hope will be of interest...

I believe that endpoints-python/endpoints/swagger_generator.py may be in need of some modification. Here's why:

I've written a protorpc message class that has some required fields on it, such as:

```
class MyClass(messages.Message):
    my_field = messages.StringField(1, required=True)
```

When I use the swagger generator, I get a swagger output similar to this:

```
"definitions": {
    "MyClass": {
        "properties": {
            "my_field": {
                "required": true,
                "type": "string"
            }
        },
        "type": "object"
    }
}
```

If it weren't for the "required" attribute, the above would work fine. However, when it is there, I get the following error when I try to gcloud beta service-management deploy the swagger spec:

> "Invalid Swagger spec. Please fix the schema errors:\n\"/definitions/
> ...
> 
> ...
> required\": domain: validation; keyword: type; message: instance has incorrect type; expected: [\"array\"]; found: \"boolean\""

Upon reviewing the OpenAPI spec, I noted that for schema objects, JSON Schema structure should be used, which places the required properties in an array.

http://json-schema.org/latest/json-schema-validation.html#rfc.section.5.15

So, I took my required properties for each of my definitions, and put them in an array, similar to this:

```
"definitions": {
    "MyClass": {
        "properties": {
            "my_field": {
                "type": "string"
            },
            "my_other_field": {
                "type": "string"
            }
        },
        "type": "object",
        "required": [
            "my_field",
            "my_other_field"
        ]
    }
}

```

Then, I tried to deploy the swagger spec again, and it worked!

So, in short, I think the swagger_generator may need change the way that it declares the required fields in the output. If there aren't any required fields defined, everything works fine. When there are, gcloud returns errors.

I'm on:
gcloud v133.0.0
google-endpoints v2.0.0b4

**Comment by bradfriedman** [comment]
Thanks for this! I'll see if I can submit a fix for the Swagger generation tool.

**Comment by Synthoova** [comment]
Brad,

Great! Thanks for doing that.

-Harley

**Comment by Synthoova** [comment]
This has been resolved. The OpenAPI specification now generates correctly.
